pkg/secrethub/client.go

```diff
@@ -46,6 +46,23 @@ func NewClient(decrypter Decrypter, authenticator auth.Authenticator, opts *Clie
 	}
 }
 
+// NewClientAWS creates a new SecretHub client that uses AWS STS and KMS to access SecretHub.
+func NewClientAWS(opts *ClientOptions) (Client, error) {
+	decrypter, err := aws.NewKMSDecrypter()
+	if err != nil {
+		return nil, err
+	}
+	client := &clientAdapter{
+		client: newClient(decrypter, auth.NopAuthenticator{}, opts),
+	}
+	authenticator, err := client.Auth().AWS().Authenticate()
+	if err != nil {
+		return nil, err
+	}
+	client.client.httpClient.authenticator = authenticator
```

Comment:
Haha this is a weird assignment. Necessary though because of the injection thing.